provider/pkg/provider/purrl/purrl.go

```diff
@@ -131,10 +132,10 @@ func callAPIEndpoint(method, url, body *string, responseCode *[]string, headers
 	}
 
 	if !responseCodeChecker(stringConversionList, code) {
-		return nil, errors.New("response code not in list of expected response codes")
+		return strPtr(string(resp.StatusCode)), nil, errors.New("response code not in list of expected response codes")
 	}
 
-	return strPtr(string(respBody)), nil
+	return strPtr(string(resp.StatusCode)), strPtr(string(respBody)), nil
```

Comment:
Create a new function:

```go
func intPtr(i int) *int {
	return &i
}
```

and change to `return intPtr(resp.StatusCode), ...